### ISSUE TITLE
Package usage questions

### ISSUE THREAD

**Issue by HDVinnie** [cybercog/laravel-ban#3]
Im just trying to see if this package is truly what im looking for. Maybe I can get some insight. With package am I (ADMIN) able to ban users from accessing my site in a way where they can still login but instead of seeing the index they will instead see a custom banned message or something along those lines. 

Also if possible am I able to ban users from say there profile page in which I can select a time limit of ban, and comment area to explain why they were banned in which would be displayed like said above.

Lastly is there any sample views controllers on how this can be accomplished with this package?

**Comment by a-komarev** [comment]
Yes. There is a [Middleware](https://github.com/cybercog/laravel-ban/blob/master/src/Http/Middleware/ForbidBannedUser.php#L51) which returns user on the previous page and pushing error message in the session. You can use it, or write your own if you need to override the logic, error key or message.

You can ban user for a limited time and write a reason of this ban:
```php
$user->ban([
    'comment' => 'You was banned for a 2 days because you was rude in a chat!',
    'expired_at' => '+2 days',
]);
```
If you want to show the reason of ban to user only once (for example send email to user with ban information) you can listen for the `\Cog\Ban\Events\ModelWasBanned` event and use `$ban->comment` value for the notification.
If you want to show the reason of the ban each time on user login you can get latest ban model for the user in your custom Middleware and store it in the flash message.
```php
auth()->user()->bans->first();
```

There are no samples yet.

**Comment by HDVinnie** [comment]
thanks for the info.......any plans to add samples?

**Comment by a-komarev** [comment]
@HDVinnie not now. Feel free to make a PR or write a Wiki pages.

Not all planned functionality was implemented.

**Comment by a-komarev** [comment]
There is a [Laravel Ban usage tutorial](http://itsolutionstuff.com/post/how-to-create-ban-revoke-user-functionality-in-laravel-5-example-example.html) where code samples could be found.
